Framework/Kernel/inc/MantidKernel/Cache.h

```diff
@@ -74,7 +99,26 @@ template <class KEYTYPE, class VALUETYPE> class DLLExport Cache {
    * @param value The new value for the key
    */
   void setCache(const KEYTYPE &key, const VALUETYPE &value) {
-    m_cacheMap[key] = value;
+    //
+    auto where = m_cacheMap.find(key);
```

Comment:
Interesting link, thanks! So apparently even two threads can insert at the same time, and one of the insertions would be dropped afterwards? What I am not sure about is whether or not *modification* of values is safe.